src/coprocessor/endpoint.rs

```diff
@@ -214,7 +215,9 @@ impl<E: Engine> Endpoint<E> {
         let (callback, future) = tikv_util::future::paired_future_callback();
         let val = engine.async_snapshot(ctx, callback);
         future::result(val)
-            .and_then(|_| future.map_err(|cancel| storage::kv::Error::Other(box_err!(cancel))))
+            .and_then(|_| {
+                future.map_err(|cancel| KvError(box KvErrorInner::Other(box_err!(cancel))))
```

Comment:
prefer to use `KvErrorInner::Other(...).into()` instead of `box KvErrorInner...` so that you don't need the `box_syntax` feature gate.